test/ecto/query/planner_test.exs

```diff
@@ -848,11 +848,23 @@ defmodule Ecto.Query.PlannerTest do
     assert_raise Ecto.QueryError, ~r/could not find named binding `as\(:posts\)`/, fn ->
       from(Post, where: as(^as).code == ^123) |> normalize()
     end
+  end
+
+  test "normalize: creating dynamic bindings with as" do
+    as = [:posts]
```

Comment:
Use a `{:posts}` in the test because it is invalid AST and it may help find bugs. :)

Reply:
Will do 👌